File: exports/lipus_module/frontend/lipus.ts
import { apiFetch } from './client';

export interface LipusDetails {
    id: string;
    muayene_id: string;
    takip_donemi: string;
    ed_tedavisi_6ay?: string;
    pde5_yaniti?: string;
    alerji_var?: boolean;
    cerrahi_oyku?: string;
    eslik_eden_hastalik?: string;
    kullanilan_ilaclar?: string;
    iief_s1?: number;
    iief_s2?: number;
    iief_s3?: number;
    iief_s4?: number;
    iief_s5?: number;
    iief_s6?: number;
    iief_total?: number;
    sep2?: string;
    sep3?: string;
    gaq1?: string;
    gaq2?: string;
    ehs_skor?: number;
    memnuniyet_sabah?: number;
    memnuniyet_cinsel?: number;
    memnuniyet_mast?: number;
    vas_skor?: number;
    yan_etki_kizariklik?: boolean;
    yan_etki_morarma?: boolean;
    yan_etki_hematuri?: boolean;
    yan_etki_yanma?: boolean;
    yan_etki_diger?: string;
    created_at?: string;
    updated_at?: string;
}

export const lipusApi = {
    getDashboardData: async (patientId: string) => {
        return apiFetch(`/api/v1/clinical/lipus/patients/${patientId}/dashboard`);
    },
    
    getDetailsByMuayene: async (muayeneId: string) => {
        return apiFetch(`/api/v1/clinical/lipus/muayene/${muayeneId}`);
    },

    createDetails: async (data: Partial<LipusDetails>) => {
        return apiFetch('/api/v1/clinical/lipus', {
            method: 'POST',
            body: JSON.stringify(data),
        });
    },

    updateDetails: async (id: string, data: Partial<LipusDetails>) => {
        return apiFetch(`/api/v1/clinical/lipus/${id}`, {
            method: 'PUT',
            body: JSON.stringify(data),
        });
    },

    deleteDetails: async (id: string) => {
        return apiFetch(`/api/v1/clinical/lipus/${id}`, {
            method: 'DELETE',
        });
    }
};

File: backend/app/schemas/clinical/lipus.py
from typing import Optional, List
from uuid import UUID
from pydantic import BaseModel, ConfigDict
from datetime import datetime


class LipusIIEFSchema(BaseModel):
    """
    Isolated 5-question IIEF schema for Lipus WBL-ED tracking.
    """
    iief_s1: Optional[int] = None
    iief_s2: Optional[int] = None
    iief_s3: Optional[int] = None
    iief_s4: Optional[int] = None
    iief_s5: Optional[int] = None
    iief_s6: Optional[int] = None
    iief_total: Optional[int] = None


class LipusDetailsBase(BaseModel):
    muayene_id: UUID
    takip_donemi: str  # Tarama, 0. Hafta, 4. Hafta, 8. Hafta, 12. Hafta

    # Tıbbi Özgeçmiş
    ed_tedavisi_6ay: Optional[str] = None
    pde5_yaniti: Optional[str] = None
    pde5_kullanim: Optional[str] = None
    ek_tedavi: Optional[str] = None
    alerji_var: Optional[bool] = False
    cerrahi_oyku: Optional[str] = None
    eslik_eden_hastalik: Optional[str] = None
    kullanilan_ilaclar: Optional[str] = None

    # Klinik Anketler (Mapped from individual columns)
    iief_s1: Optional[int] = None
    iief_s2: Optional[int] = None
    iief_s3: Optional[int] = None
    iief_s4: Optional[int] = None
    iief_s5: Optional[int] = None
    iief_s6: Optional[int] = None
    iief_total: Optional[int] = None

    sep2: Optional[str] = None
    sep3: Optional[str] = None
    gaq1: Optional[str] = None
    gaq2: Optional[str] = None
    ehs_skor: Optional[int] = None

    memnuniyet_sabah: Optional[int] = None
    memnuniyet_cinsel: Optional[int] = None
    memnuniyet_mast: Optional[int] = None

    # Güvenlik
    vas_skor: Optional[int] = None
    yan_etki_kizariklik: Optional[bool] = False
    yan_etki_morarma: Optional[bool] = False
    yan_etki_hematuri: Optional[bool] = False
    yan_etki_yanma: Optional[bool] = False
    yan_etki_diger: Optional[str] = None


class LipusDetailsCreate(LipusDetailsBase):
    pass


class LipusDetailsUpdate(BaseModel):
    takip_donemi: Optional[str] = None
    ed_tedavisi_6ay: Optional[str] = None
    pde5_yaniti: Optional[str] = None
    pde5_kullanim: Optional[str] = None
    ek_tedavi: Optional[str] = None
    alerji_var: Optional[bool] = None
    cerrahi_oyku: Optional[str] = None
    eslik_eden_hastalik: Optional[str] = None
    kullanilan_ilaclar: Optional[str] = None
    iief_s1: Optional[int] = None
    iief_s2: Optional[int] = None
    iief_s3: Optional[int] = None
    iief_s4: Optional[int] = None
    iief_s5: Optional[int] = None
    iief_s6: Optional[int] = None
    iief_total: Optional[int] = None
    sep2: Optional[str] = None
    sep3: Optional[str] = None
    gaq1: Optional[str] = None
    gaq2: Optional[str] = None
    ehs_skor: Optional[int] = None
    memnuniyet_sabah: Optional[int] = None
    memnuniyet_cinsel: Optional[int] = None
    memnuniyet_mast: Optional[int] = None
    vas_skor: Optional[int] = None
    yan_etki_kizariklik: Optional[bool] = None
    yan_etki_morarma: Optional[bool] = None
    yan_etki_hematuri: Optional[bool] = None
    yan_etki_yanma: Optional[bool] = None
    yan_etki_diger: Optional[str] = None


class LipusDetailsResponse(LipusDetailsBase):
    id: UUID
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class LipusDashboardItem(LipusDetailsBase):
    """
    Schema for the trend dashboard items containing full details and exam date.
    """
    id: UUID
    tarih: datetime
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)

